CMakeLists.txt

```diff
@@ -1,141 +1,32 @@
-
 list(APPEND CMAKE_MODULE_PATH
-     "${CMAKE_CURRENT_LIST_DIR}/cmake"
-     "${CMAKE_CURRENT_LIST_DIR}/cmake/packages"
-     "${CMAKE_CURRENT_LIST_DIR}/cmake/modules/share/modules/cmake")
-
-include(CMConfig)
-include(CMSetupVersion)
-
-if(NOT CMAKE_WORKSPACE_NAME OR NOT ("${CMAKE_WORKSPACE_NAME}" STREQUAL "crypto3"))
-    cm_workspace(crypto3)
-endif()
-
-option(ZK_PLACEHOLDER_PROFILING "Build with placeholder profiling" FALSE)
-option(ZK_PLACEHOLDER_DEBUG "Build with placeholder debug output" FALSE)
-
-if(ZK_PLACEHOLDER_PROFILING)
-    add_definitions(-DZK_PLACEHOLDER_PROFILING_ENABLED)
-endif()
-
-if(ZK_PLACEHOLDER_DEBUG)
-    add_definitions(-DZK_PLACEHOLDER_DEBUG_ENABLED)
-endif()
-
-option(BLUEPRINT_DEBUG "Build with blueprint debug output" FALSE)
-
-if(BLUEPRINT_DEBUG)
-    message(STATUS "BLUEPRINT DEBUG enabled")
-    add_definitions(-DBLUEPRINT_DEBUG_ENABLED)
-endif()
-
-option(BLUEPRINT_TESTS_PROFILING "Build tests with blueprint table profiling" FALSE)
-option(BLUEPRINT_PLACEHOLDER_PROOF_GEN "Build with placeholder proof generation running" FALSE)
-
-if(BLUEPRINT_TESTS_PROFILING)
-    message(STATUS "BLUEPRINT TESTS PROFILING enabled")
-    add_definitions(-DBLUEPRINT_PLONK_PROFILING_ENABLED)
-endif()
-
-if(BLUEPRINT_PLACEHOLDER_PROOF_GEN)
-    message(STATUS "BLUEPRINT PROOF GENERATION enabled")
-    add_definitions(-DBLUEPRINT_PLACEHOLDER_PROOF_GEN_ENABLED)
-endif()
-
-cm_project(blueprint WORKSPACE_NAME ${CMAKE_WORKSPACE_NAME} LANGUAGES C CXX)
+     "${CMAKE_CURRENT_LIST_DIR}/cmake/modules/share/modules/cmake"
+)
 
-include(CMDeploy)
-include(FindPkgConfig)
+project(crypto3_blueprint LANGUAGES C CXX)
+set(BLUEPRINT_VERSION 0.1.0)
 
 option(CMAKE_ENABLE_TESTS "Enable tests" FALSE) # used by CMTest module
 option(BUILD_EXAMPLES "Build examples" FALSE)
 
 find_package(crypto3 REQUIRED)
 find_package(Boost REQUIRED COMPONENTS container random filesystem log log_setup program_options thread system)
-if(CMAKE_ENABLE_TESTS)
-    find_package(Boost REQUIRED COMPONENTS unit_test_framework timer)
-endif()
-
-
-if(UNIX AND BUILD_WITH_PROCPS)
-    find_package(Procps)
-endif()
 
-list(APPEND ${CURRENT_PROJECT_NAME}_PUBLIC_HEADERS
-    include/nil/crypto3/zk/components/algebra/curves/edwards/r1cs/element_g1.hpp
-    include/nil/crypto3/zk/components/algebra/curves/weierstrass/r1cs/element_g1.hpp
-    include/nil/crypto3/zk/components/algebra/curves/weierstrass/r1cs/element_g2.hpp
-    include/nil/crypto3/zk/components/algebra/curves/detail/r1cs/scalar_mul.hpp
 
-    include/nil/crypto3/zk/components/algebra/fields/r1cs/exponentiation.hpp
-    include/nil/crypto3/zk/components/algebra/fields/r1cs/element_fp2.hpp
-    include/nil/crypto3/zk/components/algebra/fields/r1cs/element_fp3.hpp
-    include/nil/crypto3/zk/components/algebra/fields/r1cs/element_fp4.hpp
-    include/nil/crypto3/zk/components/algebra/fields/r1cs/element_fp6.hpp
+add_library(blueprint INTERFACE)
+add_library(crypto3::blueprint ALIAS blueprint)
```

Comment:
The target will be exported with namespace `crypto3::`, so adding alias target for consistency.

Reply:
Same question - isn't it the time to move away from `crypto3` prefix in blueprint?